### DISCUSSION TITLE
Is it necessary to add configuration to AndroidManifest.xml?

### DISCUSSION THREAD

**Discussion by lanistor** [getsentry/sentry-dart#255] [Q&A] [answered]
In Sentry's guide, we need to add configuration to AndroidManifest.xml for catching Android's errors, such as:

```xml
<application
        tools:replace="android:label"
        android:name=".Application"
        android:label="@string/title"
        android:icon="@mipmap/app_icon">
  <meta-data android:name="io.sentry.dsn" android:value="https://xxx@yyy.ingest.sentry.io/5563670" />
  ...
</application>
```

Is this step necessary while using sentry_flutter?

**Comment by bruno-garcia** [comment, marked as answer]
No. The Sentry Flutter SDK initializes the Android layer for you (passing down the options that you've written in Dart). So you don't need to configure it in the `AndroidManifest.xml`

**Comment by kuhnroyal** [comment]
How/When is this done? If any errors occur before the FlutterEngine is started, do they get reported?
Or errors that occur in native code without a FlutterEngine present (NSUrlSession, Service etc.)?

**Reply by bruno-garcia** [reply]
Errors that happen in the native layer get captured by Sentry, since the Dart SDK initializes the native layer that captures it all. 
If a crash happens in Flutter itself to the point that Flutter can't even call the `main` method in Dart, then that's not captured. That said, that's very unlikely and would probably mean a bug in Flutter itself. It's possible to capture such errors by initializing the native SDK separately but that means duplicating all the options/configuration and logic in the Dart and native layers. Also might require some changes in the Dart side because today it assumes it is the one initializing the native layer (which isn't designed to be reinitialized, think native signal handler, etc).

**Comment by kuhnroyal** [comment]
@bruno-garcia I am currently still using the flutter_sentry plugin which allows me to catch things like this https://github.com/flutter/flutter/issues/32756 - That would not be possible to find with the 4.x version.
These failures usually indicate a problem somewhere in the build but they are hard to find without the Sentry logs.
I also have logging from background services which don't have a running FlutterEngine so the errors are not sent, but would they be stored locally until the app is started again?

**Reply by bruno-garcia** [reply]
That's a fair point and it has come up before on React Native. The trade off here is the ease of configuration vs being able to capture such errors. You can still "go manual" here. Might need a toggle like we added in RN: `enable-native-plugin`. /cc @marandaneto as we discussed this a bit today. Would you mind raising an issue so we can track this?
